# 2025-04-22

## 1. 이해 안되는거 정리


#### UserView

```python
serializer = UserSerializer(data=request.data)
if serializer.is_valid():
    serializer.save() # 여기서 시리얼라이저의 create가 호출됨 => 딘순히 내부적으로 모델의 save를 호출함
```
* serializer.save()는 DRF의 인터페이스 메서드로, 내부적으로 create(), update()를 호출함

#### UserSerializer

```python
class UserSerializer(...):
    def create(self, validated_data):
        user = User.objects.create_user(**validated_data)
        return user # 여기서 유저 만들어서 리턴됨
```

#### UserModel

User.object.create_user로 인해서 유저 모델이 호출 - 유저모델 안에서 UserManager()를 호출 - create_user가 호출되는 것

```python
def create_user(...)
    user = self.model(...)
    user.set_password()
    user.save(usering=self._db) # 여기에서 직접 ㄷㅣ비에 저장됨
    return user
```
* user.save() - 모델 인스턴스를 실제 디비에 저장

## 2. ValidationError, ValueError

## 3. ModelViewSet 까보기
